### PR TITLE
3347 Create child information screen

### PR DESCRIPTION
**Pull request by Fbasham**
### Description
Add screen B-9

NOTE: 
- translations are NOT in. 
- the adult-child apply flow hasn't been implemented in main (yet), therefore logic in this PR will have to change in a future PR

### Related Azure Boards Work Items
[AB#3347](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3347)

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+
+  if (expectedCsrfToken !== submittedCsrfToken) {
+    log.warn('Invalid CSRF token detected; expected: [%s], submitted: [%s]', expectedCsrfToken, submittedCsrfToken);
+    throw new Response('Invalid CSRF token', { status: 400 });
+  }
+
+  const formAction = z.nativeEnum(FormAction).parse(formData.get('_action'));
+
+  if (formAction === FormAction.Cancel) {
+    invariant(state.adultState.childInformation, 'Expected state.applicantInformation to be defined');
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx:89]:
```suggestion
    invariant(state.adultState.childInformation, 'Expected state.childInformation to be defined');
```

File: frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+  const { csrfToken, defaultState, editMode } = useLoaderData<typeof loader>();
+  const params = useParams();
+  const fetcher = useFetcher<typeof action>();
+  const isSubmitting = fetcher.state !== 'idle';
+  const errorSummaryId = 'error-summary';
+  const [hasSocialInsuranceNumberValue, setHasSocialInsuranceNumberValue] = useState(defaultState?.hasSocialInsuranceNumber ?? '');
+
+  const handleOnPreferredMethodChecked: ChangeEventHandler<HTMLInputElement> = (e) => {
+    setHasSocialInsuranceNumberValue(e.target.value);
+  };
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx:205]: **Nitpick:** This function name can be misleading as we're dealing with SIN selection. Maybe something like `handleSocialInsuranceNumberSelection` (or similar) would be more appropriate?

File: frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+        .number({
+          required_error: t('apply:child-information.error-message.date-of-birth-day-required'),
+          invalid_type_error: t('apply:child-information.error-message.date-of-birth-day-number'),
+        })
+        .int()
+        .positive(),
+      dateOfBirth: z.string(),
+      hasSocialInsuranceNumber: z.string().trim().min(1),
+      socialInsuranceNumber: z.string().trim().optional(),
+      isParent: z.string().trim().min(1),
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx:123]: Should `hasSocialInsuranceNumber` and `isParent` have an error message attached in this validation schema?

Comment (Fbasham) [reply frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx]: Yes, although what those are is yet to be determined.  Added in some translations.

File: frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx (hunk 2 above)
Comment (Fbasham) [reply frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx]: Guess you now know I copied and pasted from somewhere ;)

File: frontend/app/route-helpers/apply-adult-route-helpers.server.ts [hunk 4]
```diff
@@ -33,6 +34,7 @@ export interface ApplyAdultState {
   readonly disabilityTaxCredit?: DisabilityTaxCreditState;
   readonly livingIndependently?: LivingIndependentlyState;
   readonly allChildrenUnder18?: AllChildrenUnder18State;
+  readonly childInformation?: ChildInformationState;
```
Comment (sebastien-comeau) [inline frontend/app/route-helpers/apply-adult-route-helpers.server.ts:37]: It should reside in the `adultChildState` but this doesn't exists at the moment. Please work on this with @AdamP-HRSDC-RHDCC  and @shawn320.

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**